src/Validation/Validation.php

```diff
@@ -838,7 +895,55 @@ public static function enum(mixed $check, string $enumClassName): bool
             return false;
         }
 
-        return $enumClassName::tryFrom($check) !== null;
+        $options += [
+            'only' => null,
+            'except' => null,
+        ];
+
+        $enum = $enumClassName::tryFrom($check);
+        if ($enum === null) {
+            return false;
+        }
+
+        return static::isValidEnum($enum, $options);
+    }
+
+    /**
+     * @param \BackedEnum $enum
+     * @param array<string, mixed> $options
+     * @return bool
+     */
+    protected static function isValidEnum(BackedEnum $enum, array $options): bool
+    {
+        if ($options['only']) {
+            if (!is_array($options['only'])) {
+                $options['only'] = [$options['only']];
```

Comment:
You can just cast to an array:
```
$only = (array)$options['only'];
```

Reply:
I tried that, and then it would cast the enum to an array afaik.

Reply:
Oh, I see. Doesn't work with an object.